src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddFileMetadataToDeposit.scala

```diff
@@ -17,32 +17,100 @@ package nl.knaw.dans.easy.multideposit.actions
 
 import java.io.File
 
+import nl.knaw.dans.common.lang.dataset.AccessCategory
 import nl.knaw.dans.easy.multideposit.actions.AddFileMetadataToDeposit._
 import nl.knaw.dans.easy.multideposit.{ Settings, UnitAction, _ }
-import nl.knaw.dans.lib.error.TraversableTryExtensions
+import nl.knaw.dans.lib.error.{ CompositeException, TraversableTryExtensions }
 import org.apache.tika.Tika
 
 import scala.util.control.NonFatal
-import scala.util.{ Failure, Try }
+import scala.util.{ Failure, Success, Try }
 import scala.xml.Elem
 
 case class AddFileMetadataToDeposit(row: Int, entry: (DatasetID, Dataset))(implicit settings: Settings) extends UnitAction[Unit] {
 
   val (datasetID, dataset) = entry
 
-  // TODO @rvanheest move the mimetype computation to a separate action once the new
-  // parameterized version of Action is installed?
-  lazy val mimetypeMap: Try[List[(File, MimeType)]] = Try {
+  case class FileInstruction(file: File, title: Option[String], subtitles: Seq[Subtitle])
+
+  sealed abstract class AudioVideo(val vocabulary: String)
+  case object Audio extends AudioVideo("http://schema.org/AudioObject")
+  case object Video extends AudioVideo("http://schema.org/VideoObject")
+
+  case class Subtitle(filepath: File, language: Option[String])
+
+  class FileMetadata(val filepath: File, val mimeType: MimeType)
+  case class AVFileMetadata(override val filepath: File,
+                            override val mimeType: MimeType,
+                            vocabulary: AudioVideo,
+                            title: String,
+                            accessibleTo: FileAccessRights.UserCategory,
+                            subtitles: Seq[Subtitle]) extends FileMetadata(filepath, mimeType)
+
+  private lazy val fileInstructions: Map[File, FileInstruction] = {
+    def resolve(filepath: String): File = new File(settings.multidepositDir, filepath).getAbsoluteFile
+
+    dataset.getColumns("AV_FILE", "AV_FILE_TITLE", "AV_SUBTITLES", "AV_SUBTITLES_LANGUAGE")
+      .toRows
+      .map(row => (
+        row.get("AV_FILE"),
+        row.get("AV_FILE_TITLE"),
+        row.get("AV_SUBTITLES"),
+        row.get("AV_SUBTITLES_LANGUAGE")
+      ))
+      .collect { case (file, title, sub, subLang) if file.exists(!_.isBlank) => (resolve(file.get), title, sub.map(resolve), subLang) }
+      .groupBy { case (file, _, _, _) => file }
+      .map { case (file, instr) =>
+        val optTitle = instr.collectFirst { case (_, Some(title), _, _) => title }
+        val subtitles = instr.collect { case (_, _, Some(sub), subLang) => Subtitle(sub, subLang) }
```

Comment:
I also saw maybe as prefix, which in my opinion could be as wel a try as an opt, so indeed I prefer the prefix opt.